internal/app_config.go

```diff
@@ -0,0 +1,127 @@
+package internal
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/pelletier/go-toml"
+)
+
+var DEF_CONFIG_ORDER = []string{
+	"/usr/lib/sota/conf.d",
+	"/var/sota/sota.toml",
+	"/etc/sota/conf.d/",
+}
+
+type cfgFile struct {
+	name string
+	path string
+
+	tree *toml.Tree
+}
+
+type AppConfig struct {
+	cfgs []*cfgFile
+}
+
+// NewAppConfig parses config files as per:
+// https://docs.ota.here.com/ota-client/latest/aktualizr-config-options.html#_how_toml_files_are_processed
+func NewAppConfig(configPaths []string) (*AppConfig, error) {
+	configsMap := make(map[string]string)
+
+	// Build up the files
+	for _, path := range configPaths {
+		if st, err := os.Stat(path); err == nil {
+			if st.IsDir() {
+				if entries, err := os.ReadDir(path); err == nil {
+					for _, entry := range entries {
+						if strings.HasSuffix(entry.Name(), ".toml") {
+							configsMap[entry.Name()] = filepath.Join(path, entry.Name())
+						}
+					}
+				}
+			} else {
+				configsMap[st.Name()] = path
```

Comment:
I wonder if we need to check if `st.Name()` has the `.toml` extension?

Reply:
In this case the file was explicitly set by the caller. e.g. something like `fioconfig -c /var/sota.toml-andy checkin`. I think we should honor that.